src/cogent3/app/composable.py

```diff
@@ -1011,8 +1011,8 @@ def define_app(klass=None, *, app_type: AppType = GENERIC, composable: bool = Tr
 
     app_type = AppType(app_type)
 
-    if inspect.isfunction(klass):
-        klass = _class_from_func(klass)
+#    if inspect.isfunction(klass):
```

Comment:
this should not be commented out

Reply:
I had commented it. I don't know why you moved this part of code into `wrapped` function.

Reply:
meant to be duplicated, not moved

Reply:
```python
def decorate_app(klass=None, /, ...):
    if klass.is function(convert to a class)
    def wrap(klass):
         if klass.is function(convert to a class)
```
the reason for both is the first one is the case where
```python
@decorate_app
def foo(): ...
```
the second one is
```python
@decorate_app(composable=False) # for example
def foo(): ...
```

Reply:
I think both above cases run the first `if klass.isfunction` code.
But the first one is called when defining and decorating the class and the second one is called when creating an instance of class.